### PR TITLE
Include `Which` for `HomebrewUserWrapper`

### LINKED ISSUE #183: undefined method `which' for #<HomebrewUserWrapper:0x000000010b9f6a58>
## :ghost: Brief Description

undefined method `which' for #<HomebrewUserWrapper:0x000000010b9f6a58>

## :pancakes: Cookbook version

5.4.5

## :woman_cook:  Chef-Infra Version

18.3.0

## :tophat: Platform details

 platform=mac_os_x
 platform_version=14.3

## Steps To Reproduce

converge cookbook

## :police_car: Expected behavior

execute normally

## :heavy_plus_sign: Additional context

```
  NoMethodError
  -------------
  undefined method `which' for #<HomebrewUserWrapper:0x000000010b9f6a58>

  Cookbook Trace: (most recent call first)
  ----------------------------------------
    /var/folders/8r/pjq314jj7mzfr_fd98_5cvbr0000gq/T/cookbooks.PdJuE9l5Vh/site-cookbooks/homebrew/libraries/helpers.rb:64:in `owner'
    /var/folders/8r/pjq314jj7mzfr_fd98_5cvbr0000gq/T/cookbooks.PdJuE9l5Vh/site-cookbooks/homebrew/recipes/default.rb:63:in `block in from_file'
    /var/folders/8r/pjq314jj7mzfr_fd98_5cvbr0000gq/T/cookbooks.PdJuE9l5Vh/site-cookbooks/homebrew/recipes/default.rb:61:in `from_file'

  Relevant File Content:
  ----------------------
  /var/folders/8r/pjq314jj7mzfr_fd98_5cvbr0000gq/T/cookbooks.PdJuE9l5Vh/site-cookbooks/homebrew/libraries/helpers.rb:

   57:      ::File.exist?("#{HomebrewWrapper.new.install_path}/bin/brew")
   58:    end
   59:
   60:    def owner
   61:      @owner ||= begin
   62:        # once we only support 14.0 we can switch this to find_homebrew_username
   63:        require 'etc'
   64>>       ::Etc.getpwuid(HomebrewUserWrapper.new.find_homebrew_uid).name
   65:                 rescue Chef::Exceptions::CannotDetermineHomebrewOwner
   66:                   calculate_owner
   67:      end.tap do |owner|
   68:        Chef::Log.debug("Homebrew owner is #{owner}")
   69:      end
   70:    end
   71:
   72:    private
   73:

  System Info:
  ------------
  chef_version=18.3.0
  platform=mac_os_x
  platform_version=14.3
  ruby=ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin20]
  program_name=/opt/chef-workstation/bin/chef-solo
  executable=/opt/chef-workstation/bin/chef-solo
```

### PR DESCRIPTION
**Pull request by jazaval**
# Description

Explicitly include `Which` module from `Chef` which fixes cookbook use on newer 18.x clients. 

## Issues Resolved

Fixes #183 

## Check List

- [✅] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [❌] New functionality includes testing.
- [❌] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 This file is used to list changes made in each version of the homebrew cookbook.
 
-## Unreleased
+## 5.4.6 - *2024-05-06*
```
Comment (Stromweld) [inline CHANGELOG.md:5]: don't include the version bump, sous-chefs automation takes care of that. Just need the ##Unreleased header and description of changes under it.

File: metadata.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 maintainer_email 'help@sous-chefs.org'
 license          'Apache-2.0'
 description      'Install Homebrew and includes resources for working with taps and casks'
-version          '5.4.5'
+version          '5.4.6'
```
Comment (Stromweld) [inline metadata.rb:6]: sous-chefs automation will handle version bumping the CB version here.

**Review by Stromweld — CHANGES_REQUESTED** [review]

**Comment by Stromweld** [comment]
Users should be using the built in homebrew resources for chef-client > 14. Not sure what this fix is needed for?

**Comment by jazaval** [comment]
Chef client won't install homebrew itself, which is generally the core remaining use here (default recipe).

**Review by Stromweld — APPROVED** [review]
LGTM

**Comment by kitchen-porter** [comment]
Released as: [5.4.8](https://github.com/sous-chefs/homebrew/releases/tag/5.4.8)

**Pull request merged**